py/server/deephaven/table_listener.py

```diff
@@ -329,16 +333,10 @@ def start(self, do_replay: bool = False, replay_lock: Literal["shared", "exclusi
             raise RuntimeError("Attempting to start an already started listener..")
 
         try:
-            def _start():
-                if do_replay:
-                    self.listener.replay()
-
-                self.t.j_table.addUpdateListener(self.listener)
-
             if do_replay:
-                _do_locked(self.t, _start, lock_type=replay_lock)
-            else:
-                _start()
+                _do_locked(self.t, self.listener_adapter.replay, lock_type=replay_lock)
```

Comment:
See my comments in the merged case. `replay_lock` should be deprecated and ignored.